pkg/graph/resolvers/trb_admin_note.go

```diff
@@ -0,0 +1,106 @@
+package resolvers
+
+import (
+	"context"
+	"errors"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+// CreateTRBAdminNote creates a new TRB admin note in the database
+func CreateTRBAdminNote(ctx context.Context, store *storage.Store, trbRequestID uuid.UUID, category models.TRBAdminNoteCategory, noteText string) (*models.TRBAdminNote, error) {
+	noteToCreate := models.TRBAdminNote{
+		TRBRequestID: trbRequestID,
+		Category:     category,
+		NoteText:     noteText,
+	}
+	noteToCreate.CreatedBy = appcontext.Principal(ctx).ID()
+
+	createdNote, err := store.CreateTRBAdminNote(ctx, &noteToCreate)
+	if err != nil {
+		return nil, err
+	}
+
+	return createdNote, nil
+}
+
+// GetTRBAdminNoteByID retrieves a single admin note by its ID
+func GetTRBAdminNoteByID(ctx context.Context, store *storage.Store, id uuid.UUID) (*models.TRBAdminNote, error) {
+	note, err := store.GetTRBAdminNoteByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	if note == nil {
+		return nil, &apperrors.ResourceNotFoundError{
+			Err:      err,
+			Resource: models.TRBAdminNote{},
+		}
+	}
+
+	return note, nil
+}
+
+// GetTRBAdminNotesByTRBRequestID retrieves a list of admin notes associated with a TRB request
+func GetTRBAdminNotesByTRBRequestID(ctx context.Context, store *storage.Store, trbRequestID uuid.UUID) ([]*models.TRBAdminNote, error) {
+	notes, err := store.GetTRBAdminNotesByTRBRequestID(ctx, trbRequestID)
+	if err != nil {
+		return nil, err
+	}
+
+	return notes, nil
+}
+
+// GetTRBAdminNoteAuthorInfo gets the full user info for the author of an admin note
+func GetTRBAdminNoteAuthorInfo(ctx context.Context, authorEUAID string, fetchUserInfo func(context.Context, string) (*models.UserInfo, error)) (*models.UserInfo, error) {
+	authorInfo, err := fetchUserInfo(ctx, authorEUAID)
```

Comment:
This seems like a case where we could skip having a function in the resolvers package, since all you're doing is passing a function into the resolvers package so it can call that function and return the results.

Reply:
Yeah, I was on the fence about separating it out when I wrote it; inlining it in `schema.resolvers.go` probably makes more sense.

Reply:
Inlined in https://github.com/CMSgov/easi-app/pull/1879/commits/3a602d1344bc57d2597513bfda620580ba0b3ecb.